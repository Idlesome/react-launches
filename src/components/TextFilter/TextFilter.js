/**
 * Basic material text input field that allows for customisation of
 * label and input change event
 */
import React from "react";
import PropTypes from "prop-types";

import { makeStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";

const useStyles = makeStyles({
  marginLeft: "2vmin",
  marginRight: "2vmin",
  width: 200
});

function TextFilter(props) {
  const { onChange, label } = props;

  const classes = useStyles();

  return (
    <TextField
      label={label}
      type="filter"
      className={classes.root}
      margin="normal"
      onChange={onChange}
    />
  );
}
TextFilter.propTypes = {
  onChange: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired
};

export default TextFilter;
