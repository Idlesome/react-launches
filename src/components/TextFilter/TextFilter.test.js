import React from "react";

import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import TextFilter from "./TextFilter";

Enzyme.configure({ adapter: new Adapter() });

// TODO: test onChange function is called
describe("<LaunchCard />", () => {
  it("should render component", () => {
    const wrapper = mount(
      <TextFilter label="Text filter" onChange={() => {}} />
    );
    expect(wrapper.find("input")).toMatchSnapshot();
  });
});
