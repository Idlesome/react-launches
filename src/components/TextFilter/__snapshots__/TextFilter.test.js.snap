// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LaunchCard /> should render component 1`] = `ReactWrapper {}`;
